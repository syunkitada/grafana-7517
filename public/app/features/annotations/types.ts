import { PanelData, DataFrame, AnnotationEvent, TimeRange } from '@grafana/data';
import { DashboardModel, PanelModel } from '../dashboard/state';

export interface AnnotationQueryOptions {
  dashboard: DashboardModel;
  panel: PanelModel;
  range: TimeRange;
}

export interface AnnotationQueryResponse {
  /**
   * All the data flattened to a single frame
   */
  frame?: DataFrame;

  /**
   * The processed annotation events
   */
  events?: AnnotationEvent[];

  /**
   * The original panel response
   */
  panelData?: PanelData;
}
