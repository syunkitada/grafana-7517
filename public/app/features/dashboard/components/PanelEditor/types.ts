import { PanelModel } from '../../state/PanelModel';

export enum DisplayMode {
  Fill = 0,
  Fit = 1,
  Exact = 2,
}

export const displayModes = [
  { value: DisplayMode.Fill, label: 'Fill', description: 'Use all avaliable space' },
  { value: DisplayMode.Fit, label: 'Fit', description: 'Fit in the space keeping ratio' },
  { value: DisplayMode.Exact, label: 'Exact', description: 'Same size as the dashboard' },
];

export enum EditorTab {
  Query = 'query',
  Alerts = 'alerts',
  Transform = 'xform',
}

export const allTabs = [
  { tab: EditorTab.Query, label: 'Query', show: (panel: PanelModel) => true },
  { tab: EditorTab.Alerts, label: 'Alerts', show: (panel: PanelModel) => true },
  { tab: EditorTab.Transform, label: 'Transform', show: (panel: PanelModel) => true },
];
