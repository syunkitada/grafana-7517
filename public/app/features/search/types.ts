export enum DashboardSearchItemType {
  DashDB = 'dash-db',
  DashHome = 'dash-home',
  DashFolder = 'dash-folder',
}

export interface DashboardSection {
  id: number;
  uid?: string;
  title: string;
  expanded?: boolean;
  url: string;
  icon: string;
  score: number;
  hideHeader?: boolean;
  checked?: boolean;
  items: DashboardSectionItem[];
  toggle?: (section: DashboardSection) => Promise<DashboardSection>;
  selected?: boolean;
  type: DashboardSearchItemType;
}

export interface DashboardSectionItem {
  checked?: boolean;
  folderId?: number;
  folderTitle?: string;
  folderUid?: string;
  folderUrl?: string;
  id: number;
  isStarred: boolean;
  selected?: boolean;
  tags: string[];
  title: string;
  type: DashboardSearchItemType;
  uid: string;
  uri: string;
  url: string;
}

export interface DashboardTag {
  term: string;
  count: number;
}

export interface DashboardQuery {
  query: string;
  mode: string;
  tag: string[];
  starred: boolean;
  skipRecent: boolean;
  skipStarred: boolean;
  folderIds: number[];
}

export interface SectionsState {
  sections: DashboardSection[];
  allChecked: boolean;
  dashboardTags: DashboardTag[];
}

export type ItemClickWithEvent = (item: DashboardSectionItem | DashboardSection, event: any) => void;

export type SearchAction = {
  type: string;
  payload?: any;
};

export interface OpenSearchParams {
  query?: string;
}
