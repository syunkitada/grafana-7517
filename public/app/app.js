define([
  'angular',
  'jquery',
  'lodash',
  'app/core/config',
  'require',
  'bootstrap',
  'angular-route',
  'angular-sanitize',
  'angular-strap',
  'angular-dragdrop',
  'angular-ui',
  'bindonce',
  'app/core/core',
],
function (angular, $, _, config, appLevelRequire) {
  "use strict";

  var app = angular.module('grafana', []);
  var register_fns = {};
  var preBootModules = [];

  // This stores the grafana version number
  app.constant('grafanaVersion',"@grafanaVersion@");

  /**
   * Tells the application to watch the module, once bootstraping has completed
   * the modules controller, service, etc. functions will be overwritten to register directly
   * with this application.
   * @param  {[type]} module [description]
   * @return {[type]}        [description]
   */
  app.useModule = function (module) {
    if (preBootModules) {
      preBootModules.push(module);
    } else {
      _.extend(module, register_fns);
    }
    // push it into the apps dependencies
    apps_deps.push(module.name);
    return module;
  };

  app.config(function($locationProvider, $controllerProvider, $compileProvider, $filterProvider, $provide) {
    register_fns.controller = $controllerProvider.register;
    register_fns.directive  = $compileProvider.directive;
    register_fns.factory    = $provide.factory;
    register_fns.service    = $provide.service;
    register_fns.filter     = $filterProvider.register;
  });

  var apps_deps = [
    'grafana.core',
    'ngRoute',
    'ngSanitize',
    '$strap.directives',
    'ang-drag-drop',
    'grafana',
    'pasvaz.bindonce',
    'ui.bootstrap',
    'ui.bootstrap.tpls',
  ];

  var module_types = ['controllers', 'directives', 'factories', 'services', 'filters', 'routes'];

  _.each(module_types, function (type) {
    var module_name = 'grafana.'+type;
    // create the module
    app.useModule(angular.module(module_name, []));
  });

  var preBootRequires = ['app/features/all'];
  var pluginModules = config.bootData.pluginModules || [];

  // add plugin modules
  for (var i = 0; i < pluginModules.length; i++) {
    preBootRequires.push(pluginModules[i]);
  }

  app.boot = function() {
    require(preBootRequires, function () {

      // disable tool tip animation
      $.fn.tooltip.defaults.animation = false;

      // bootstrap the app
      angular
        .element(document)
        .ready(function() {
          angular.bootstrap(document, apps_deps)
            .invoke(['$rootScope', function ($rootScope) {
              _.each(preBootModules, function (module) {
                _.extend(module, register_fns);
              });

              preBootModules = null;

              $rootScope.requireContext = appLevelRequire;
              $rootScope.require = function (deps, fn) {
                var $scope = this;
                $scope.requireContext(deps, function () {
                  var deps = _.toArray(arguments);
                  fn.apply($scope, deps);
                });
              };
            }]);
        });
    });
  };

  return app;
});
