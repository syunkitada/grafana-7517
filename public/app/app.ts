///<reference path="headers/common.d.ts" />

///<amd-dependency path="bootstrap" />
///<amd-dependency path="angular-strap" />
///<amd-dependency path="angular-route" />
///<amd-dependency path="angular-sanitize" />
///<amd-dependency path="angular-dragdrop" />
///<amd-dependency path="angular-bindonce" />
///<amd-dependency path="angular-ui" />
///<amd-dependency path="app/core/core" />

import 'es6-shim';
import 'es6-promise';

import _ = require('lodash');
import $ = require('jquery');
import bootstrap = require('bootstrap');
import kbn = require('app/core/utils/kbn');
import angular = require('angular');
import config = require('app/core/config');

class GrafanaApp {
  register_fns: any = {};

  useModule(module) {
    _.extend(module, this.register_fns);
    return module;
  }

  init() {
    var app = angular.module('grafana', []);
    app.constant('grafanaVersion', "@grafanaVersion@");

    app.config(($locationProvider, $controllerProvider, $compileProvider, $filterProvider, $provide) => {
      console.log('app config');
      this.register_fns.controller = $controllerProvider.register;
      this.register_fns.directive  = $compileProvider.directive;
      this.register_fns.factory    = $provide.factory;
      this.register_fns.service    = $provide.service;
      this.register_fns.filter     = $filterProvider.register;
    });

    var apps_deps = [
      'grafana.core',
      'ngRoute',
      'ngSanitize',
      '$strap.directives',
      'ang-drag-drop',
      'grafana',
      'pasvaz.bindonce',
      'ui.bootstrap.tabs',
    ];

    var module_types = ['controllers', 'directives', 'factories', 'services', 'filters', 'routes'];

    _.each(module_types, type => {
      var module_name = 'grafana.' + type;
      this.useModule(angular.module(module_name, []));
      apps_deps.push(module_name);
    });

    var preBootRequires = [System.import('app/features/all')];
    var pluginModules = config.bootData.pluginModules || [];

    // add plugin modules
    for (var i = 0; i < pluginModules.length; i++) {
      preBootRequires.push(System.import(pluginModules[i]));
    }

    Promise.all(preBootRequires).then(function() {
      // disable tool tip animation
      $.fn.tooltip.defaults.animation = false;
      // bootstrap the app
      var asd = angular.bootstrap(document, apps_deps).invoke(['$rootScope', function ($rootScope) {
        console.log('bootstrap');
      }]);
    }).catch(function(err) {
      console.log('Application boot failed: ' + err);
    });
  }
 }

 var grafanaApp = new GrafanaApp();

 export = {
   init: function() {
     grafanaApp.init();
   },
   useModule: function(m) {
     grafanaApp.useModule(m);
   }
 };
