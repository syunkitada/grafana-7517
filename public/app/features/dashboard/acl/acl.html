<div class="modal-body">
  <div class="modal-header">
    <h2 class="modal-header-title">
      <i class="fa fa-lock"></i>
      <span class="p-l-1">Permissions</span>
    </h2>

    <a class="modal-header-close" ng-click="ctrl.dismiss();">
      <i class="fa fa-remove"></i>
    </a>
  </div>

  <div class="modal-content">
    <table class="filter-table gf-form-group">
      <tr ng-repeat="acl in ctrl.items" ng-class="{'gf-form-disabled': acl.inherited}">
        <td style="width: 100%;">
          <i class="{{acl.icon}}"></i>
          <span ng-bind-html="acl.nameHtml"></span>
        </td>
        <td>
          <em class="muted no-wrap" ng-show="acl.inherited">Inherited from folder</em>
        </td>
        <td class="query-keyword">Can</td>
        <td>
          <div class="gf-form-select-wrapper">
            <select class="gf-form-input gf-size-auto" ng-model="acl.permission" ng-options="p.value as p.text for p in ctrl.permissionOptions" ng-change="ctrl.permissionChanged(acl)" ng-disabled="acl.inherited"></select>
          </div>
        </td>
        <td>
          <a class="btn btn-inverse btn-small" ng-click="ctrl.removeItem($index)" ng-hide="acl.inherited">
            <i class="fa fa-remove"></i>
          </a>
        </td>
      </tr>
      <tr ng-show="ctrl.aclItems.length === 0">
        <td colspan="4">
          <em>No permissions. Will only be accessible by admins.</em>
        </td>
      </tr>
    </table>

    <div class="gf-form-inline">
      <form name="addPermission" class="gf-form-group">
        <h6 class="muted">Add Permission For</h6>
        <div class="gf-form-inline">
          <div class="gf-form">
            <div class="gf-form-select-wrapper">
              <select class="gf-form-input gf-size-auto" ng-model="ctrl.newType" ng-options="p.value as p.text for p in ctrl.aclTypes"  ng-change="ctrl.typeChanged()"></select>
            </div>
          </div>
          <div class="gf-form" ng-show="ctrl.newType === 'User'">
            <user-picker user-picked="ctrl.userPicked($user)"></user-picker>
          </div>
          <div class="gf-form" ng-show="ctrl.newType === 'Group'">
            <user-group-picker user-group-picked="ctrl.groupPicked($group)"></user-group-picker>
          </div>
        </div>
      </form>
      <div class="gf-form width-17">
        <span ng-if="ctrl.error" class="text-error p-l-1">
          <i class="fa fa-warning"></i>
          {{ctrl.error}}
        </span>
      </div>
    </div>

    <div class="gf-form-button-row text-center">
      <button type="button" class="btn btn-danger" ng-disabled="!ctrl.canUpdate" ng-click="ctrl.update()">
        Update Permissions
      </button>
      <a class="btn-text" ng-click="ctrl.dismiss();">Close</a>
    </div>
  </div>
</div>

  <!-- <br> -->
  <!-- <br> -->
  <!-- <br> -->
  <!--  -->
  <!-- <div class="permissionlist"> -->
  <!--   <div class="permissionlist__section"> -->
  <!--     <div class="permissionlist__section&#45;header"> -->
  <!--       <h6>Permissions</h6> -->
  <!--     </div> -->
  <!--     <table class="filter&#45;table form&#45;inline"> -->
  <!--       <thead> -->
  <!--         <tr> -->
  <!--           <th style="width: 50px;"></th> -->
  <!--           <th>Name</th> -->
  <!--           <th style="width: 220px;">Permission</th> -->
  <!--           <th style="width: 120px"></th> -->
  <!--         </tr> -->
  <!--       </thead> -->
  <!--       <tbody> -->
  <!--         <tr ng&#45;repeat="permission in ctrl.userPermissions" class="permissionlist__item"> -->
  <!--           <td><i class="fa fa&#45;fw fa&#45;user"></i></td> -->
  <!--           <td>{{permission.userLogin}}</td> -->
  <!--           <td class="text&#45;right"> -->
  <!--             <a ng&#45;click="ctrl.removePermission(permission)" class="btn btn&#45;danger btn&#45;small"> -->
  <!--               <i class="fa fa&#45;remove"></i> -->
  <!--             </a> -->
  <!--           </td> -->
  <!--         </tr> -->
  <!--         <tr ng&#45;repeat="permission in ctrl.userGroupPermissions" class="permissionlist__item"> -->
  <!--           <td><i class="fa fa&#45;fw fa&#45;users"></i></td> -->
  <!--           <td>{{permission.userGroup}}</td> -->
  <!--           <td><select class="gf&#45;form&#45;input gf&#45;size&#45;auto" ng&#45;model="permission.permissions" ng&#45;options="p.value as p.text for p in ctrl.permissionTypeOptions" ng&#45;change="ctrl.updatePermission(permission)"></select></td> -->
  <!--           <td class="text&#45;right"> -->
  <!--             <a ng&#45;click="ctrl.removePermission(permission)" class="btn btn&#45;danger btn&#45;small"> -->
  <!--               <i class="fa fa&#45;remove"></i> -->
  <!--             </a> -->
  <!--           </td> -->
  <!--         </tr> -->
  <!--         <tr ng&#45;repeat="role in ctrl.roles" class="permissionlist__item"> -->
  <!--           <td></td> -->
  <!--           <td>{{role.name}}</td> -->
  <!--           <td><select class="gf&#45;form&#45;input gf&#45;size&#45;auto" ng&#45;model="role.permissions" ng&#45;options="p.value as p.text for p in ctrl.roleOptions" ng&#45;change="ctrl.updatePermission(role)"></select></td> -->
  <!--           <td class="text&#45;right"> -->
  <!--  -->
  <!--           </td> -->
  <!--         </tr> -->
  <!--       </tbody> -->
  <!--     </table> -->
  <!--   </div> -->
  <!--   </div> -->
  <!-- </div> -->
