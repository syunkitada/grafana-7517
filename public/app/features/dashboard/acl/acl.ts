///<reference path="../../../headers/common.d.ts" />

import coreModule from 'app/core/core_module';
import appEvents from 'app/core/app_events';
import _ from 'lodash';

export class AclCtrl {
  dashboard: any;
  userAcl: DashboardAcl[];
  groupAcl: DashboardAcl[];
  permissionTypeOptions = [
    {value: 1, text: 'View'},
    {value: 2, text: 'Edit'},
    {value: 4, text: 'Admin'}
  ];

  roleOptions = [
    {value: 0, text: 'No Access'},
    {value: 1, text: 'View'},
    {value: 2, text: 'Edit'},
    {value: 4, text: 'Admin'}
  ];

  roles = [];

  type = 'User Group';
  permission = 1;
  userId: number;
  userGroupId: number;

  /** @ngInject */
  constructor(private backendSrv, private dashboardSrv) {
    this.userAcl = [];
    this.groupAcl = [];
    this.dashboard = dashboardSrv.getCurrent();
    this.get(this.dashboard.id);
  }

  get(dashboardId: number) {
    return this.backendSrv.get(`/api/dashboards/id/${dashboardId}/acl`)
      .then(result => {
        this.userAcl = _.filter(result, p => { return p.userId > 0;});
        this.groupAcl = _.filter(result, p => { return p.userGroupId > 0;});
        this.roles = this.setRoles(result);
      });
  }

  setRoles(result: any) {
    return [
      {name: 'Viewer', permissions: 1},
      {name: 'Editor', permissions: 2},
      {name: 'Admin', permissions: 4}
    ];
  }

  addPermission() {
    if (this.type === 'User') {
      if (!this.userId) {
        return;
      }
      return this.addOrUpdateUserPermission(this.userId, this.permission).then(() => {
        this.userId = null;
        return this.get(this.dashboard.id);
      });
    } else {
      if (!this.userGroupId) {
        return;
      }

      return this.addOrUpdateUserGroupPermission(this.userGroupId, this.permission).then(() => {
        this.userGroupId = null;
        return this.get(this.dashboard.id);
      });
    }
  }

  addOrUpdateUserPermission(userId: number, permissions: number) {
    return this.backendSrv.post(`/api/dashboards/id/${this.dashboard.id}/acl`, {
      userId: userId,
      permissions: permissions
    });
  }

  addOrUpdateUserGroupPermission(userGroupId: number, permissions: number) {
    return this.backendSrv.post(`/api/dashboards/id/${this.dashboard.id}/acl`, {
      userGroupId: userGroupId,
      permissions: permissions
    });
  }

  updatePermission(permission: DashboardAcl) {
    if (permission.userId > 0) {
      return this.addOrUpdateUserPermission(permission.userId, permission.permissions);
    } else {
      if (!permission.userGroupId) {
        return;
      }
      return this.addOrUpdateUserGroupPermission(permission.userGroupId, permission.permissions);
    }
  }

  removePermission(permission: DashboardAcl) {
    return this.backendSrv.delete(`/api/dashboards/id/${permission.dashboardId}/acl/${permission.id}`).then(() => {
      return this.get(permission.dashboardId);
    });
  }
}

export function dashAclModal() {
  return {
    restrict: 'E',
    templateUrl: 'public/app/features/dashboard/acl/acl.html',
    controller: AclCtrl,
    bindToController: true,
    controllerAs: 'ctrl',
    scope: {
      dismiss: "&"
    }
  };
}

export interface FormModel {
  dashboardId: number;
  userId?: number;
  userGroupId?: number;
  PermissionType: number;
}

export interface DashboardAcl {
  id: number;
  orgId: number;
  dashboardId: number;
  created: Date;
  updated: Date;
  userId: number;
  userLogin: number;
  userEmail: string;
  userGroupId: number;
  userGroup: string;
  permissions: number;
  permissionName: string;
}

coreModule.directive('dashAclModal', dashAclModal);
