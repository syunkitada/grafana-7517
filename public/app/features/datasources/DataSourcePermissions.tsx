import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import SlideDown from '../../core/components/Animations/SlideDown';
import AddDataSourcePermissions from './AddDataSourcePermissions';
import { AclTarget } from 'app/types/acl';
import { addDataSourcePermission, loadDataSourcePermissions, removeDataSourcePermission } from './state/actions';
import { DashboardAcl, DataSourcePermission } from 'app/types';
import { getRouteParamsId } from '../../core/selectors/location';
import PermissionList from '../../core/components/PermissionList/PermissionList';

export interface Props {
  dataSourcePermissions: DataSourcePermission[];
  pageId: number;
  addDataSourcePermission: typeof addDataSourcePermission;
  loadDataSourcePermissions: typeof loadDataSourcePermissions;
  removeDataSourcePermission: typeof removeDataSourcePermission;
}

interface State {
  isAdding: boolean;
}

export class DataSourcePermissions extends PureComponent<Props, State> {
  state = {
    isAdding: false,
  };

  componentDidMount() {
    this.fetchDataSourcePermissions();
  }

  async fetchDataSourcePermissions() {
    const { pageId, loadDataSourcePermissions } = this.props;

    return await loadDataSourcePermissions(pageId);
  }

  onOpenAddPermissions = () => {
    this.setState({
      isAdding: true,
    });
  };

  onAddPermission = state => {
    const { pageId, addDataSourcePermission } = this.props;
    const data = {
      permission: state.permission,
      userId: 0,
      teamId: 0,
    };

    if (state.type === AclTarget.Team) {
      data.teamId = state.teamId;
    } else if (state.team === AclTarget.User) {
      data.userId = state.userId;
    }

    addDataSourcePermission(pageId, data);
  };

  onRemovePermission = (item: DashboardAcl) => {
    this.props.removeDataSourcePermission(1, 1);
  };

  onCancelAddPermission = () => {
    this.setState({
      isAdding: false,
    });
  };

  render() {
    const { dataSourcePermissions } = this.props;
    const { isAdding } = this.state;

    return (
      <div>
        <div className="page-action-bar">
          <h3 className="page-sub-heading">Permissions</h3>
          <div className="page-action-bar__spacer" />
          <button className="btn btn-success pull-right" onClick={this.onOpenAddPermissions} disabled={isAdding}>
            <i className="fa fa-plus" /> Add Permission
          </button>
        </div>
        <SlideDown in={isAdding}>
          <AddDataSourcePermissions
            onAddPermission={state => this.onAddPermission(state)}
            onCancel={this.onCancelAddPermission}
          />
        </SlideDown>
        <PermissionList
          items={dataSourcePermissions}
          onRemoveItem={this.onRemovePermission}
          onPermissionChanged={() => {}}
          isFetching={false}
        />
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    pageId: getRouteParamsId(state.location),
    dataSourcePermissions: state.dataSources.dataSourcePermissions,
  };
}

const mapDispatchToProps = {
  addDataSourcePermission,
  loadDataSourcePermissions,
  removeDataSourcePermission,
};

export default connect(mapStateToProps, mapDispatchToProps)(DataSourcePermissions);
