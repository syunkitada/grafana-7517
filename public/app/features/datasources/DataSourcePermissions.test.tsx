import React from 'react';
import { shallow } from 'enzyme';
import { DataSourcePermissions, Props } from './DataSourcePermissions';
import { DataSourcePermission, DataSourcePermissionDTO } from 'app/types';
import { AclTarget, dashboardPermissionLevels } from '../../types/acl';

const setup = (propOverrides?: object) => {
  const props: Props = {
    dataSourcePermission: {} as DataSourcePermissionDTO,
    pageId: 1,
    addDataSourcePermission: jest.fn(),
    enableDataSourcePermissions: jest.fn(),
    disableDataSourcePermissions: jest.fn(),
    loadDataSourcePermissions: jest.fn(),
    removeDataSourcePermission: jest.fn(),
  };

  Object.assign(props, propOverrides);

  const wrapper = shallow(<DataSourcePermissions {...props} />);
  const instance = wrapper.instance() as DataSourcePermissions;

  return {
    wrapper,
    instance,
  };
};

describe('Render', () => {
  it('should render component', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render permissions enabled', () => {
    const { wrapper } = setup({
      dataSourcePermission: {
        enabled: true,
        datasourceId: 1,
        permissions: [] as DataSourcePermission[],
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});

describe('Functions', () => {
  describe('on add permissions', () => {
    const { instance } = setup();

    it('should add permissions for team', () => {
      const mockState = {
        permission: dashboardPermissionLevels[0].value,
        teamId: 1,
        type: AclTarget.Team,
      };

      instance.onAddPermission(mockState);

      expect(instance.props.addDataSourcePermission).toHaveBeenCalledWith(1, { teamId: 1, permission: 1 });
    });

    it('should add permissions for user', () => {
      const mockState = {
        permission: dashboardPermissionLevels[0].value,
        userId: 1,
        type: AclTarget.User,
      };

      instance.onAddPermission(mockState);

      expect(instance.props.addDataSourcePermission).toHaveBeenCalledWith(1, { userId: 1, permission: 1 });
    });
  });
});
