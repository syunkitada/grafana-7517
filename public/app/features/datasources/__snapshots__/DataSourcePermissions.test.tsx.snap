// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div>
  <div
    className="page-action-bar"
  >
    <h3
      className="page-sub-heading"
    >
      Permissions
    </h3>
    <div
      className="page-action-bar__spacer"
    />
  </div>
  <div
    className="empty-list-cta"
  >
    <div
      className="empty-list-cta__title"
    >
      Permissions not enabled for this data source.
    </div>
    <button
      className="empty-list-cta__button btn btn-xlarge btn-success"
      onClick={[Function]}
    >
      Enable
    </button>
    <div
      className="empty-list-cta__pro-tip"
    >
      <i
        className="fa fa-rocket"
      />
       ProTip:
       
      Only admins will be able to query the data source after you enable permissions.
    </div>
  </div>
</div>
`;

exports[`Render should render permissions enabled 1`] = `
<div>
  <div
    className="page-action-bar"
  >
    <h3
      className="page-sub-heading"
    >
      Permissions
    </h3>
    <div
      className="page-action-bar__spacer"
    />
    <button
      className="btn btn-success pull-right"
      disabled={false}
      key="add-permission"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
       Add Permission
    </button>
    <button
      className="btn btn-danger pull-right"
      key="disable-permissions"
      onClick={[Function]}
    >
      Disable Permissions
    </button>
  </div>
  <div>
    <Component
      in={false}
    >
      <AddDataSourcePermissions
        onAddPermission={[Function]}
        onCancel={[Function]}
      />
    </Component>
    <DataSourcePermissionsList
      items={Array []}
      onRemoveItem={[Function]}
    />
  </div>
</div>
`;
